docker/Dockerfile_txsim

```diff
@@ -0,0 +1,49 @@
+# stage 1 Generate celestia-appd Binary
+FROM docker.io/golang:1.20.5-alpine3.17 as builder
+# hadolint ignore=DL3018
+RUN apk update && apk add --no-cache \
+    gcc \
+    git \
+    make \
+    musl-dev
+COPY . /celestia-app
+WORKDIR /celestia-app
+# we need the appd build as we might want to create an account
+# internally for txsimulation
+RUN make build
+RUN make txsim-build
```

Comment:
we could unify these two `RUN` in one as we have [here](https://github.com/celestiaorg/celestia-node/blob/main/Dockerfile#L16) to keep the builds in one layer